exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,667 @@
+    assert!(status.is_err());
+    assert!(status
+        .unwrap_err()
+        .to_string()
+        .contains("java.lang.StackOverflowError;"));
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_service_error_on_exec_tx() {
+    let tx_args = &[255_u8, 2_u8, 3_u8, 4_u8];
+    let tx_error_code = tx_args[0];
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::SRVC_ERR_ON_EXEC_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    let status = block.transactions.get(0).unwrap().status();
+    assert!(status.is_err());
+    assert_service_error_with_code(status.unwrap_err(), tx_error_code);
+
+    test_kit.stop();
+}
+
+#[test]
+fn submit_failing_on_exec_tx() {
+    let tx_args = &[1_u8, 2_u8, 3_u8, 4_u8];
+
+    let (runtime, instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+
+    test_kit.create_block();
+
+    let tx = create_transaction(instance_id, fake_service::FAIL_ON_EXEC_TX, tx_args);
+    let block = test_kit.create_block_with_transaction(tx);
+
+    assert_eq!(block.transactions.len(), 1);
+    let status = block.transactions.get(0).unwrap().status();
+    assert!(status.is_err());
+    assert!(status
+        .unwrap_err()
+        .to_string()
+        .contains("java.lang.ArithmeticException;"));
+
+    test_kit.stop();
+}
+
+#[test]
+fn before_commit() {
+    let (runtime, _instance_id, config, _path) = create_runtime_with_valid_test_config();
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(vec![config])
+        .create();
+
+    let mut value: Option<i32> = None;
+
+    for _ in 0..10 {
+        test_kit.create_block();
+
+        let snapshot = test_kit.snapshot();
+        let map = create_before_commit_test_map(&*snapshot, fake_service::TEST_SERVICE_NAME);
+        let key = hash(fake_service::BEFORE_COMMIT_ENTRY_KEY.as_ref());
+        let cur_value = map
+            .get(&key)
+            .expect("Failed to find the entry created in beforeCommit")
+            .parse::<i32>()
+            .expect("Failed to parse the entry value created in beforeCommit");
+
+        if let Some(prev_value) = value {
+            assert_eq!(prev_value + 1, cur_value);
+        }
+
+        value = Some(cur_value);
+    }
+
+    test_kit.stop();
+}
+
+#[test]
+fn state_hashes() {
+    let service_hash = &[
+        Hash::from_hex("8c1ea14c7893acabde2aa95031fae57abb91516ddb78b0f6622afa0d8cb1b5c2").unwrap(),
```

Comment:
It is unclear why whatever comes
from the runtime shall match these. It seems brittle: some changes to the service
(not in the `state_hashes`, but in other methods) might cause these to change and
it won't be clear if the code is broken, or the constants.

I think it might be better to _unit_ test the conversion separately (extracting the code that
takes a byte array from Java and converts it to the appropriate return type).

In this IT it might be sensible to get the expected hashes from the collections.